scripts/Deploy-MSBuild.ps1

```diff
@@ -0,0 +1,45 @@
+[CmdletBinding(PositionalBinding=$false)]
+Param(
+  [Parameter(Mandatory = $true)]
+  [string] $destination,
+  [ValidateSet('Debug','Release')]
+  [string] $configuration = "Debug"
+)
+
+function Copy-WithBackup ($origin) {
+    $destinationPath = Join-Path -Path $destination -ChildPath (Split-Path $origin -leaf)
+
+    if (Test-Path $destinationPath -PathType Leaf) {
+        # Back up previous copy of the file
+        Copy-Item $destinationPath $BackupFolder -ErrorAction Stop
+    }
+
+    Copy-Item $origin $destinationPath -ErrorAction Stop
+}
+
+# TODO: find destination in PATH if not specified
+
+# TODO: identify processes likely to be using target MSBuild and warn/offer to kill
+
+# TODO: find most-recently-built MSBuild and make it default $configuration
+
+$BackupFolder = New-Item (Join-Path $destination -ChildPath "Backup-$(Get-Date -Format FileDateTime)") -itemType directory -ErrorAction Stop
```

Comment:
I've been using git to init and track changes of the msbuild bin directory, since it requires fewer command on my part. But not sure it's worth it when a script does it.